app/src/main/java/com/example/cinemapp/util/ThemeUtil.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.cinemapp.util
+
+import android.app.UiModeManager
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import androidx.appcompat.app.AppCompatDelegate
+
+fun setAppTheme(mode: Int, context: Context) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        val uiModeManager = context.getSystemService(Context.UI_MODE_SERVICE) as UiModeManager
```

Comment:
i'm not sure if it is needed, but i think you should consider using DI for UiModeManager.
Check if it is recommended to do so